tests/TestGIN.py

```diff
@@ -1,104 +1,154 @@
 import random
-import sys
-import io
-
-sys.path.append("")
 import unittest
 
 import numpy as np
-import matplotlib.image as mpimg
-import matplotlib.pyplot as plt
 
 from causallearn.search.HiddenCausal.GIN.GIN import GIN
 
 
 class TestGIN(unittest.TestCase):
-    def test_case1(self):
-        sample_size = 1000
-        np.random.seed(0)
-        L1 = np.random.uniform(-1, 1, size=sample_size) ** 5
-        L2 = np.random.uniform(0.5, 2.0) * L1 + np.random.uniform(-1, 1, size=sample_size) ** 5
-        X1 = np.random.uniform(0.5, 2.0) * L1 + np.random.uniform(-1, 1, size=sample_size) ** 5
-        X2 = np.random.uniform(0.5, 2.0) * L1 + np.random.uniform(-1, 1, size=sample_size) ** 5
-        X3 = np.random.uniform(0.5, 2.0) * L2 + np.random.uniform(-1, 1, size=sample_size) ** 5
-        X4 = np.random.uniform(0.5, 2.0) * L2 + np.random.uniform(-1, 1, size=sample_size) ** 5
+    def test_case1_kci(self):
+        sample_size = 500
+        random.seed(42)
+        np.random.seed(42)
+        L1 = np.random.uniform(-1, 1, size=sample_size)
+        L2 = np.random.uniform(1.2, 1.8) * L1 + np.random.uniform(-1, 1, size=sample_size)
+        X1 = np.random.uniform(1.2, 1.8) * L1 + 0.2 * np.random.uniform(-1, 1, size=sample_size)
+        X2 = np.random.uniform(1.2, 1.8) * L1 + 0.2 * np.random.uniform(-1, 1, size=sample_size)
+        X3 = np.random.uniform(1.2, 1.8) * L2 + 0.2 * np.random.uniform(-1, 1, size=sample_size)
+        X4 = np.random.uniform(1.2, 1.8) * L2 + 0.2 * np.random.uniform(-1, 1, size=sample_size)
+        data = np.array([X1, X2, X3, X4]).T
+        data = (data - np.mean(data, axis=0)) / np.std(data, axis=0)
+        _, causal_order = GIN(data, indep_test_method='kci')
+        causal_order = [sorted(cluster_i) for cluster_i in causal_order]
+        ground_truth = [[0, 1], [2, 3]]
+        assert len(causal_order) == len(ground_truth)
+        for i in range(len(causal_order)):
+            assert np.isclose(causal_order[i], ground_truth[i]).all()
```

Comment:
you don't need to use np.isclose() right?

causal_order must equal to ground_truth, right?

Reply:
For the first one, I don't understand. For the second one, yes.